numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -2588,48 +2588,7 @@ gentype_getreadbuf(PyObject *self, Py_ssize_t segment, void **ptrptr)
     return numbytes;
 }
 
-#if !defined(NPY_PY3K)
-static Py_ssize_t
-gentype_getsegcount(PyObject *self, Py_ssize_t *lenp)
-{
-    PyArray_Descr *outcode;
-
-    outcode = PyArray_DescrFromScalar(self);
-    if (lenp) {
-        *lenp = outcode->elsize;
-#ifndef Py_UNICODE_WIDE
-        if (outcode->type_num == NPY_UNICODE) {
-            *lenp >>= 1;
-        }
-#endif
-    }
-    Py_DECREF(outcode);
-    return 1;
-}
-
-static Py_ssize_t
-gentype_getcharbuf(PyObject *self, Py_ssize_t segment, constchar **ptrptr)
-{
-    if (PyArray_IsScalar(self, String) ||
-            PyArray_IsScalar(self, Unicode)) {
-        return gentype_getreadbuf(self, segment, (void **)ptrptr);
-    }
-    else {
-        PyErr_SetString(PyExc_TypeError,
-                "Non-character array cannot be interpreted "\
-                "as character buffer.");
-        return -1;
-    }
-}
-#endif /* !defined(NPY_PY3K) */
-
 static PyBufferProcs gentype_as_buffer = {
-#if !defined(NPY_PY3K)
-    gentype_getreadbuf,                          /* bf_getreadbuffer*/
```

Comment:
gentype_getreadbuf is still used in gentype_byteswap

Reply:
Can you maybe add a comment to gentype_getreadbuf like:
```
/* This function matches the Python 2.7 PyBufferProcs.bf_getreadbuffer interface, but no longer needs to.
 * In future we could consider rewriting callers to use `gentype_getbuffer`, or inline the function body at the caller.
 */
```

Reply:
Added at function definition, thanks for the well worded comment.